spanconfig: NPE in completeZoneConfig caused by missing error checking in GetHydratedZoneConfigForNamedZone

Saw the following panic randomly in a logic test: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_Test/4079518?buildTab=overview

Looking at the callstack, seems like we're just forgetting to check an error.... how come this isn't caught by a linter?

https://github.com/cockroachdb/cockroach/blob/ad27839166526a92444bf0f719cf9ec5f62c2f7f/pkg/sql/zone_config.go#L277-L283

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2fa5978]

goroutine 6063184 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc08fb2bd40, {0x4d5b5d0, 0xc131eafca0})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:251 +0x94
panic({0x3761e00, 0x6e2a810})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/sql.completeZoneConfig(0x0, {{0xc079a38df8}, {0xc079a38df8}, 0x0}, 0x251ea1a0, 0xc0251ea1a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/zone_config.go:145 +0x58
github.com/cockroachdb/cockroach/pkg/sql.GetHydratedZoneConfigForNamedZone({0x4d3f9f8, 0xc16c513170}, 0xc000000026, {{0xc079a38df8}, {0xc079a38df8}, 0x0}, {0x3c65198, 0x0})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/zone_config.go:280 +0x108
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator.(*SQLTranslator).generateSpanConfigurationsForNamedZone(0xc05e9a3b60, {0x4d3f9f8, 0xc16c513170}, 0xc16c513170, 0x80eca4a8)
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go:201 +0x70a
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator.(*SQLTranslator).generateSpanConfigurations(0xc05e9a3b60, {0x4d3f9f8, 0xc16c513170}, 0x900db860, 0xc09f4e8ae8, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go:132 +0x230
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator.(*SQLTranslator).Translate.func1({0x4d3f9f8, 0xc16c513170}, 0xc0e1f87130, 0xc000074130)
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go:98 +0x35e
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*CollectionFactory).Txn.func2({0x4d3f9f8, 0xc16c513170}, 0xc0efa544a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/txn.go:102 +0x435
github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1({0x4d3f9f8, 0xc16c513170}, 0x4db1ae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849 +0x27
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc0e1f87130, {0x4d3f9f8, 0xc16c513170}, 0xc0251eaf28)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:972 +0xb7
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x4d3f9f8, 0xc16c513170}, 0xc102c7e630, 0x130)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848 +0x5a
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0x0, {0x4d3f9f8, 0xc16c513170}, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830 +0x85
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*CollectionFactory).Txn(0xc05e9a2e40, {0x4d3f9f8, 0xc16c513170}, {0x4dd82d0, 0xc06d9b87e0}, 0xc102c7e630, 0xc0251eb0c8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/txn.go:92 +0x192
github.com/cockroachdb/cockroach/pkg/sql.DescsTxn(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:3159
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator.(*SQLTranslator).Translate(0xc0251eb5b8, {0x4d3f9f8, 0xc16c513170}, {0xc09f4e8ae8, 0x0, 0x0})
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go:65 +0xe5
github.com/cockroachdb/cockroach/pkg/spanconfig.FullTranslate(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfig.go:126
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*fullReconciler).reconcile(0xc0251eb5b8, {0x4d3f9f8, 0xc16c513170})
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:184 +0xb0
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*Reconciler).Reconcile(0xc154b19800, {0x4d3f9f8, 0xc16c513170}, {0x3982200, 0x3a92c40, 0x0}, 0x3e8ea60)
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:139 +0xc5
github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob.(*resumer).Resume(0xc0251eb738, {0x4d3f9f8, 0xc16c513170}, {0x3a92c40, 0xc0c558a020})
	/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob/job.go:44 +0x66
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0x4d5b5d0, {0x4ce6858, 0xc0e1179ff8}, {0x4d3f9f8, 0xc16c513170}, {0x3a92c40, 0xc0c558a020}, 0xc0251eb948)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1123 +0xba
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc193d95900, {0x4d5b5d0, 0xc0efa544a0}, {0x3a92c40, 0xc0c558a020}, {0x4ce6858, 0xc0e1179ff8}, 0xc18c311200, {0xc104b07ce0, 0x7}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1124 +0x6d7
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc193d95900, {0x4d3f950, 0xc199c01300}, {0x4ce6858, 0xc0e1179ff8}, 0xc18c311200, {0xc104b07ce0, 0x7}, {0xc1310e3140, 0x16})
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:401 +0x54f
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x4d5b5d0, 0xc11b311ca0})
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:323 +0xfa
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488 +0x12a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:478 +0x445
```

--- a/pkg/sql/zone_config.go
+++ b/pkg/sql/zone_config.go
@@ -277,10 +277,13 @@ func GetHydratedZoneConfigForNamedZone(
 	zoneID, zone, _, _, err := getZoneConfig(
 		codec, descpb.ID(id), getKey, false /* getInheritedDefault */, false, /* mayBeTable */
 	)
+	if err != nil {
+		return nil, err
+	}
 	if err := completeZoneConfig(zone, codec, zoneID, getKey); err != nil {
 		return nil, err
 	}
-	return zone, err
+	return zone, nil
 }
 
 // GetHydratedZoneConfigForTable returns a fully hydrated zone config for a

